### PR TITLE
[Feat][deepseek-mla] Add batch and kv_group dimensions to MLA kernels

### PR DESCRIPTION
**Pull request by stelladuyx**
## Description

Add `batch` and `kv_group` dimensions to three DeepSeek MLA kernels (`fp8_lighting_indexer`, `fp8_quant`, `topk_selector`), enabling batched execution and grouped-query attention support.

### Changes

**Kernels** (`tileops/kernels/deepseek_mla/`)
- `fp8_lighting_indexer`: Add batch/kv_group dims to all tensors; refactor inner loop to iterate over kv groups with per-group gemm and scale application; update `clean_logits_` kernel accordingly
- `fp8_quant`: Parallelize kernel grid over `(batch, seq_len_kv_blocks, kv_group)`; quantize per last dim instead of per excluded dims
- `topk_selector`: Parallelize over `(batch, seq_len, kv_group)` instead of `(batch,)`; add explicit `float32` cast before `reinterpret` in `convert_to_uint32`; add bounds checks on atomic write positions

**Ops** (`tileops/ops/`)
- `fp8_lighting_indexer`: Add `batch`/`kv_group` params; split forward into `torch_quant_forward` (5 args, does FP8 cast internally) and `tl_quant_forward` (6 args, pre-quantized); update `per_custom_dims_cast_to_fp8` to use per-token (last-dim) scaling
- `fp8_quant`: Add `batch`/`kv_group` params
- `topk_selector`: Add `seq_len_kv`/`kv_group` params

**Tests & Benchmarks**
- Update all test fixtures and benchmark classes with new dimension parameters
- Update `calculate_flops` / `calculate_memory` for batch and kv_group
- Add test case with `kv_group > 1` for fp8_quant

## Type of Change

- [x] New operator implementation
- [x] Performance improvement

## Checklist

- [x] I have run `pre-commit run --all-files` and fixed all linting issues.
- [x] I have verified that my changes pass local unit tests.
- [x] **(For new ops)** I have added the corresponding `Benchmark` class in `benchmarks/`.
- [x] **(For new ops)** I have reported benchmark results in the tracking issue.

### CODE REVIEW

File: tileops/kernels/deepseek_mla/fp8_lighting_indexer.py [hunk 1]
```diff
@@ -250,11 +279,13 @@ def supply_prog(
     ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
         self.q = q
         self.kv = kv
-        seq_len, heads, index_dim = q.shape
+        batch, seq_len, heads, index_dim = q.shape
         seq_len_kv = kv.shape[0]
```
Comment (lcy-seso) [inline tileops/kernels/deepseek_mla/fp8_lighting_indexer.py]: **[major]** `kv` now has shape `(batch, seq_len_kv, kv_group, index_dim)`, so `kv.shape[0]` is `batch`, not `seq_len_kv`. Should be `kv.shape[1]`.

File: benchmarks/ops/bench_fp8_quant.py [hunk 2]
```diff
@@ -12,11 +12,12 @@ class Fp8QuantBenchmark(BenchmarkBase):
 
     def calculate_flops(self) -> Optional[float]:
         t = self.test
-        return 2 * t.seq_len_kv * t.index_dim + t.seq_len_kv + 4 * t.seq_len_kv * t.index_dim
+        return (2 * t.batch * t.seq_len_kv * t.kv_group * t.index_dim +
+                t.batch * t.seq_len_kv * t.kv_group + 4 * t.seq_len_kv * t.kv_group * t.index_dim)
```
Comment (lcy-seso) [inline benchmarks/ops/bench_fp8_quant.py]: **[medium]** Missing `t.batch` in the last term — `4 * t.seq_len_kv * t.kv_group * t.index_dim` should be `4 * t.batch * t.seq_len_kv * t.kv_group * t.index_dim` for consistency with the other terms.

File: tileops/ops/fp8_lighting_indexer.py [hunk 3]
```diff
@@ -13,42 +13,63 @@
...
         index_k, index_k_scale = self.per_custom_dims_cast_to_fp8(index_k, (0,), False)
+
+        return self.kernel(index_q, index_k, index_k_scale, weights, cu_seqlen_ks, cu_seqlen_ke)
+
+    def tl_quant_forward(self, index_q: torch.Tensor, index_k: torch.Tensor,
+                         index_k_scale: torch.Tensor, weights: torch.Tensor,
+                         cu_seqlen_ks: torch.Tensor, cu_seqlen_ke: torch.Tensor):
         return self.kernel(index_q, index_k, index_k_scale, weights, cu_seqlen_ks, cu_seqlen_ke)
 
+    def forward(self, *args) -> torch.Tensor:
```
Comment (lcy-seso) [inline tileops/ops/fp8_lighting_indexer.py]: **[medium]** Dispatching on `len(args)` is fragile and loses type safety. Consider using keyword arguments (e.g. `index_k_scale=None`) or keeping `torch_quant_forward`/`tl_quant_forward` as the public API instead of routing through `*args`.

File: tests/ops/test_fp8_lighting_indexer.py [hunk 4]
```diff
@@ -147,31 +156,59 @@ def generate_random_cu_seqlens(self,
...
+            dtype=torch.bfloat16)
+        IndexK = torch.randn(
+            self.batch,
+            self.seq_len_kv,
+            self.kv_group,
+            self.index_dim,
+            device='cuda',
+            dtype=torch.bfloat16)
         Weights = torch.randn(self.seq_len, self.heads, device='cuda', dtype=self.accum_dtype)
+        CuSeqLenKS = torch.zeros(self.seq_len, device='cuda', dtype=self.index_dtype)
```
Comment (lcy-seso) [inline tests/ops/test_fp8_lighting_indexer.py]: **[minor]** Dead code — `CuSeqLenKS` and `CuSeqLenKE` are initialized here with zeros/full, then immediately overwritten by `generate_random_cu_seqlens` on the next line. Remove the dead initialization.

File: tileops/kernels/deepseek_mla/fp8_lighting_indexer.py (hunk 1 above)
Comment (stelladuyx) [reply tileops/kernels/deepseek_mla/fp8_lighting_indexer.py]: Thank you for pointing out the errors and for your helpful suggestions.I have made the corresponding revisions.

**Review by lcy-seso — APPROVED** [review]
